### ISSUE TITLE
Method name "MyController::my_action" is not in camel caps format

### ISSUE THREAD

**Issue by davehensley** [cakephp/cakephp-codesniffer#137]
When using the CakePHP standard, phpcs complains that my action names are underscored rather than camelCased.

This seems to have been fixed a long time ago: https://github.com/cakephp/cakephp-codesniffer/issues/2

Any idea why I'm still getting the error?

**Comment by markstory** [comment]
What version of the cakephp-codesniffer are you using?

**Comment by davehensley** [comment]
PHP_CodeSniffer version 2.3.3 (stable) by Squiz (http://www.squiz.net)
cakephp/cakephp-codesniffer           2.0.4  CakePHP CodeSniffer Standards

**Comment by ADmad** [comment]
Which is the sniff which reports this error? (Use the `-p` flag when running phpcs)

**Comment by davehensley** [comment]
I think you mean -s instead of -p?


    $ ./vendor/bin/phpcs -s --standard=CakePHP src/Controller/MyController.php 
    
    FILE: .../src/Controller/MyController.php
    ----------------------------------------------------------------------
    FOUND 1 ERROR AFFECTING 1 LINE
    ----------------------------------------------------------------------
     13 | ERROR | Method name "MyController::my_action" is not in camel caps
        |       | format (PSR1.Methods.CamelCapsMethodName.NotCamelCaps)
    ----------------------------------------------------------------------
     
    Time: 44ms; Memory: 5Mb

```php
<?php
// src/Controller/MyController.php
namespace App\Controller;

class MyController extends AppController
{

    /**
     * My Action
     *
     * @return void
     */
    public function my_action()
    {
    }
}
```

**Comment by ADmad** [comment]
Yes sorry, i forgot which switch it was :)  As I had expected the requirement of camel caps method name is encored by PSR standard. Code sniffer 2.x is an extension of PSR2 (which included PSR1) hence the error is expected.

I would suggest just using camel caps method names for controller actions too and use routing to map required url format to the method. For e.g. if you use `DashedRoute` it already maps dashed url fragment `my-action` to method name `myAction()`.

**Comment by davehensley** [comment]
Thanks for the tip about DashedRoute. But are underscored action names no longer the preferred style? The book is still full of them: http://book.cakephp.org/3.0/en/controllers.html

**Comment by ADmad** [comment]
The book still uses underscored examples for historical reasons. In 3.0 you have option to use either `InflectedRoute` (which uses underscore) or `DashedRoute` (or any custom one). I would suggest using `DashedRoute` as Google recommends using dashes in urls. In 3.1 `DashedRoute` will be the default route.

**Comment by davehensley** [comment]
Ok, will do. Thanks for your help.
